collect.py

```diff
@@ -156,14 +152,13 @@ def do_something(driver, visited_paths, current_path, elem_properties=None):
             # Get all the properties of the child.
             elem_properties = get_element_properties(driver, child)
 
-            # we check if the path has been visited previously
-            if was_visited(current_path, visited_paths, elem_properties):
-                continue
-
             # If the element is not displayed or is disabled, the user can't interact with it. Skip
             # non-displayed/disabled elements, since we're trying to mimic a real user.
             if not child.is_displayed() or not child.is_enabled():
-                current_path.pop()
+                continue
+
+            # we check if the path has been visited previously else visit it
```

Comment:
`# Check if the path has been visited previously, visit it if it hasn't`

The comment at line 166 doesn't apply anymore, since we visit now.